sos/plugins/kubernetes.py

```diff
@@ -15,40 +15,130 @@
 # Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
 from sos.plugins import Plugin, RedHatPlugin
+from os import path
 
 
 class kubernetes(Plugin, RedHatPlugin):
 
     """Kubernetes plugin
     """
 
-    option_list = [("podslog", "capture logs for pods", 'slow', False)]
+    # Red Hat Atomic Platform and OpenShift Enterprise use the
+    # atomic-openshift-master package to provide kubernetes
+    packages = ('kubernetes', 'atomic-openshift-master')
+    files = ("/etc/origin/master/master-config.yaml",)
+
+    option_list = [
+        ("all", "use --all-namespaces for all kubectl commands",
+            'fast', True),
+        ("describe", "capture descriptions of all kube resources",
+            'fast', True),
+        ("podlogs", "capture logs for pods", 'slow', False),
+    ]
+
+    def check_is_master(self):
+        if any([path.exists("/var/run/kubernetes/apiserver.key"),
+                path.exists("/etc/origin/master/master-config.yaml")
+                ]):
```

Comment:
All over the code :-)

It is the most common style in use now and it is the one we recommend. I just opened a half-dozen plugins and all but `acpid.py` used this style correctly within the first few lines. It is preferred for the vertical brevity and the reduction of line length compared to starting the bracketed clause on the opening line.

If anyone wants to write up a "coding standards" document I'd happily review it and add it to the wiki but nobody has had time to date and to be honest, most of the projects that I work on do not have an explicit, written coding standard (GNU and the Linux kernel are the exceptions) - they just expect contributors to read the existing code and adopt its style.